### PR TITLE
chore: Update dependencies and add miri testing

### PR DESCRIPTION
**Pull request by luke-biel**

### CODE REVIEW

**Comment by luke-biel** [comment]
I changed semversions of crates to full ones via `cargo upgrade`. I assume automatic tool > our standards.

**Comment by luke-biel** [comment]
As addition I'm dropping dependabot. I can set a reminder for every monday and do better at updating our deps than it does, from looks of this pr.

**Comment by smores56** [comment]
So the changes themselves look good, and the removal of the dependabot is a welcome change, but I don't think we should include the patch version on all of our dependencies. It seems like it will just add a lot of annoying work when we can just leave the minor versions as-is and run `cargo update` to the same effect once a week.

**Comment by luke-biel** [comment]
> So the changes themselves look good, and the removal of the dependabot is a welcome change, but I don't think we should include the patch version on all of our dependencies. It seems like it will just add a lot of annoying work when we can just leave the minor versions as-is and run `cargo update` to the same effect once a week.

Well, I agree partialy, as we initially wanted to use `major.minor` only, but this is automated via `cargo-upgrade` command (part of `cargo-edit` https://github.com/killercup/cargo-edit). I'm not sure we want to go into every file and change deps manually one by one - that's error prone. So I'd stick with whatever format `cargo-edit` choose.

File: .github/workflows/test-unit.yml [hunk 1]
```diff
@@ -32,7 +32,7 @@ jobs:
         with:
           profile: minimal
           override: true
-          components: clippy, rustfmt
+          components: clippy, rustfmt, miri
```
Comment (Kononnable) [inline .github/workflows/test-unit.yml:35]: How is it related?

I think PR name needs updating to include all major changes.

**Review by Kononnable — APPROVED** [review]

File: .github/workflows/test-unit.yml (hunk 1 above)
Comment (luke-biel) [reply .github/workflows/test-unit.yml:35]: It's a minor change that didn't deserve separate PR, but here you go, changed the name

**Pull request merged**